lib/nine_box.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:ui';
+
+import 'package:flame/palette.dart';
+import 'package:flame/sprite.dart';
+
+/// This allows you to create a rectangle textured with a 9-sliced image.
+///
+/// How it works is that you have a template image in a 3x3 grid, made up of 9 tiles,
+/// and a new rectangle can be draw by keeping the 4 corners, expanding the 4 sides only
+/// in the direction in which they are located and expanding the center in both directions.
+/// That allows you to have non distored borders.
+class NineBox {
+  /// The sprite used to render the box, must be a 3x3 grid of square tiles.
+  Sprite sprite;
+
+  /// The size of each tile in the source sprite image.
+  int tileSize;
+
+  /// The size each tile becomes when rendered (optionally used to scale the src image).
+  int destTileSize;
+
+  /// Creates a nine-box instance.
+  ///
+  /// [srite] is the 3x3 grid and [tileSize] is the size of each tile.
+  /// The src sprite must a square of size 3*[tileSize].
+  ///
+  /// If [tileSize] is not provided, the width of the sprite is assumed as the size.
+  /// Otherwise the width and height properties of the sprite are ignored.
+  ///
+  /// If [destTileSize] is not provided, the evaluated [tileSize] is used instead
+  /// (so no scaling happens).
+  NineBox(this.sprite, {int tileSize, int destTileSize}) {
+    this.tileSize = tileSize ?? sprite.src.width.toInt();
+    this.destTileSize = destTileSize ?? tileSize;
+  }
+
+  /// Renders this nine box as a rectangle of coordinates ([x], [y]) and size ([width], [height]).
+  void draw(Canvas c, double x, double y, double width, double height) {
```

Comment:
We should probably have a drawRect as well, what do you think?

Reply:
excellent idea